README.md

```diff
@@ -4,4 +4,113 @@
 [![Hackage](https://img.shields.io/hackage/v/extensions.svg?logo=haskell)](https://hackage.haskell.org/package/extensions)
 [![MPL-2.0 license](https://img.shields.io/badge/license-MPL--2.0-blue.svg)](LICENSE)
 
-Parse Haskell Language Extensions
+Library and CLI tool to parse Haskell `{-# LANGUAGE #-}` extensions in
+source files, extract `default-extensions` from `.cabal` files and
+combine together `default-extensions` and per-module extensions.
+
+## Goals
+
+The `extensions` library provides a lighweight way to get Haskell
+LANGUAGE pragmas for Haskell modules. It has the following goals:
+
+1. Be **lightweight**. Dependency footprint is extremely small,
+   and using `extensions` either as a library or as a tool is
+   straightforward.
+2. Support both `default-extensions` in **Cabal** and `{-# LANGUAGE #-}`
+   pragmas in **Haskell** modules.
+3. Should work on common and real cases. `extensions` strives to
+   support as many valid syntactic constructions as possible, but it
+   may not work on every possible combination of CPP, comments and
+   pragmas, where GHC would work.
```

Comment:
For such cases, should we encourage people to open bug reports?

Reply:
@vrom911 Excellent idea :+1: